text/0060-refashion-updates-apis.md

```diff
@@ -0,0 +1,416 @@
+    - The possible values for the `type` field are reworked to be more clear and consistent with our naming rules.
+    - A `details` object is added to contain specific information related to a `task` payload that was previously displayed in the `type` nested object.
+    - An `indexUid` field is added to give information about the related index on which the task is performed.
+    - `duration` format has been updated to express an `ISO 8601` duration.
+    - `processed` status changes to `succeeded`.
+    - `startedProcessingAt` is updated to `startedAt`.
+    - `processedAt` is updated to `finishedAt`.
+- `202 Accepted` requests previously returning an `updateId` are now returning a summarized `task` object.
+- `MEILI_MAX_UDB_SIZE` env var is updated `MEILI_MAX_TASK_DB_SIZE`.
+- `--max-udb-size` cli option is updated to `--max-task-db-size`.
+
+
+### II. Motivation
+
+The main motivation is to stabilize the current `update` resource to a version that conforms to our API convention and thus allow future evolutions on a more solid base. We would like to modify the name `update`, its format is also to be reviewed because some attributes are not immediately clear, either in the possible values or in the chosen names.
+
+### III. Explanation
+
+#### 1. `task` object definition
+
+##### **Fully Qualified `task` object**
+
+> This fully qualified version appears as response object on `task` dedicated endpoints.
+
+| field   | type    | description                     |
+|---------|---------|---------------------------------|
+| uid      | integer | Unique sequential identifier           |
+| indexUid | string | Unique index identifier |
+| status  | string  | Status of the task. Possible values are `enqueued`, `processing`, `succeeded`, `failed`                                |
+| type    | string  | Type of the task. Possible values are `documentsAddition`, `documentsPartial`, `documentsDeletion`, `settingsUpdate`, `clearAll` |
+| details | object |  Details information of the task payload. See `details` object definition by `type` part. |
+| error | object | Error object containing error details and context when a task has a `failed` status. See https://github.com/meilisearch/specifications/pull/61|
+| duration | string | Total elasped time the engine was in processing state expressed as a `ISO-8601` duration format. Default is set to `null`.  |
+| enqueuedAt | string | Represent the date and time as `ISO-8601` format when the task has been enqueued |
+| startedAt | string | Represent the date and time as `ISO-8601` format when the task has been dequeued and started to be processed. Default is set to `null`|
+| finishedAt | string | Represent the date and time as `ISO-8601` format when the task has `failed` or `succeeded`. Default is set to `null` |
+
+> 💡 The order of the fields must be returned in this order.
```

Comment:
Don't oppose this in any way, but wanted to know why? 

I thought that, generally speaking, many/most languages discourage programmers from ever relying on the order of object fields. Or at least super high-level languages (e.g. JS, Ruby, PHP).

Reply:
Indeed, it is true that it is not recommended to rely on the order of the information of a JSON object programmatically to access the information. Access by keys is recommended and allows to be completely agnostic about the order of the fields in JSON format.

However, despite the JSON specification which does not guarantee a consistent order of fields, we have control over this information in the responses we generate on the server-side, it's considered a good practice to organize objects in a consistent way when possible. The goal here is also to make life easy for a human reading the API responses. For example, it's more pleasant to find the IDs at the top of the response when it's read by someone.

The idea is to write an API for machines but also for humans when possible.

Reply:
Makes sense, thanks for the explanation =)